plugins/modules/keycloak_authz_permission.py

```diff
@@ -0,0 +1,471 @@
+      and an admin user would work, as would a separate realm definition with the scope tailored
+      to your needs and a user having the expected roles.
+
+    - The names of module options are snake_cased versions of the camelCase options used by Keycloak.
+      The Authorization Services paths and payloads have not officially been documented by the Keycloak project.
+      https://www.puppeteers.net/blog/keycloak-authorization-services-rest-api-paths-and-payload/
+
+attributes:
+    check_mode:
+        support: none
+    diff_mode:
+        support: none
+
+options:
+    state:
+        description:
+            - State of the authorization permission.
+            - On C(present), the authorization permission will be created (or updated if it exists already).
+            - On C(absent), the authorization permission will be removed if it exists.
+        choices: ['present', 'absent']
+        default: 'present'
+        type: str
+    name:
+        description:
+            - Name of the authorization permission to create.
+        type: str
+        required: true
+    description:
+        description:
+            - The description of the authorization permission.
+        type: str
+        required: false
+    permission_type:
+        description:
+            - The type of authorization permission.
+            - On C(scope) create a scope-based permission.
+            - On C(resource) create a resource-based permission.
+        type: str
+        required: true
+        choices:
+            - resource
+            - scope
+    decision_strategy:
+        description:
+            - The decision strategy to use with this permission.
+        type: str
+        default: UNANIMOUS
+        required: false
+        choices:
+            - UNANIMOUS
+            - AFFIRMATIVE
+            - CONSENSUS
+    resources:
+        description:
+            - Resource names to attach to this permission.
+            - Scope-based permissions can only include one resource.
+            - Resource-based permissions can include multiple resources.
+        type: list
+        elements: str
+        default: []
+        required: false
+    scopes:
+        description:
+            - Scope names to attach to this permission.
+            - Resource-based permissions can't have scopes attached to them.
+        type: list
+        elements: str
+        default: []
+        required: false
+    policies:
+        description:
+            - Policy names to attach to this permission.
+        type: list
+        elements: str
+        default: []
+        required: false
+    query_state:
+        description:
+            - Query the authorization permission and add it to the result object.
+            - This adds an extra Keycloak Admin API call for each authorization permission.
+            - This is intended mainly for integration tests, not for general use.
```

Comment:
I'm not sure that this option should exist. Such functionality should be provided by an `_info` module, not by the module itself.

Reply:
Do you have any pointers / examples on how to create an info module for this kind of use-case? This is the first time I heard about it :smile:.

Reply:
See https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_general.html#creating-an-info-or-a-facts-module for example.